code/model/modelanimation.h

```diff
@@ -145,81 +150,85 @@ namespace animation {
 		}
 	};
 
-	class ModelAnimationSegment {
-	protected:
-
-		std::map<int, float> m_duration;
-
-	public:
-		virtual ~ModelAnimationSegment() = default;
-
-		float getDuration(int pmi_id) const;
-
-		//Will be called to give the animations an opportunity to recalculate based on current ship data, as well as animation data up to that point.
-		virtual void recalculate(const submodel_instance* submodel_instance, const bsp_info* submodel, const ModelAnimationData<>& base, int pmi_id) = 0;
-		//This function needs to contain anything that manipulates ModelAnimationData (such as any movement)
-		virtual ModelAnimationData<true> calculateAnimation(const ModelAnimationData<>& base, float time, int pmi_id) const = 0;
-		//This function needs to contain any animation parts that do not change ModelAnimationData (such as sound or particles)
-		virtual void executeAnimation(const ModelAnimationData<>& state, float timeboundLower, float timeboundUpper, ModelAnimationDirection direction, int pmi_id) = 0;
-	};
-	
+	class ModelAnimationSet;
 
 	class ModelAnimationSubmodel {
 	protected:
 		SCP_string m_name;
 		optional<int> m_submodel;
-		
-	private:
-		//Polymodel Instance ID + submodel ID -> ModelAnimationData
-		static std::map<std::pair<int, int>, ModelAnimationData<>> s_initialData;
+		bool is_turret = false;
 
-		std::shared_ptr<ModelAnimationSegment> m_mainSegment;
+	private:
+		//Polymodel Instance ID -> ModelAnimationData
+		std::map<int, ModelAnimationData<>> m_initialData;
 
-		friend class ModelAnimation;
 	public:
-		//Create a submodel animation based on the name of the submodel
-		ModelAnimationSubmodel(SCP_string submodelName, std::shared_ptr<ModelAnimationSegment> mainSegment);
-
+		ModelAnimationSubmodel(SCP_string submodelName);
```

Comment:
Clang will then complain that since the value is just stored in a member variable (and thus copied anyways), it should use call-by-value and then std::move() in the implementation (as it is currently)